cpp/src/rolling/rolling_detail.cuh

```diff
@@ -310,6 +310,43 @@ struct DeviceRollingRowNumber {
   }
 };
 
+struct agg_specific_empty_output {
+  template <typename InputType, aggregation::Kind op>
+  std::unique_ptr<column> operator()(column_view const& input, rolling_aggregation const& agg) const
+  {
+    using target_type = cudf::detail::target_type_t<InputType, op>;
+
+    if constexpr (std::is_same_v<cudf::detail::target_type_t<InputType, op>, void>) {
+      CUDF_FAIL("Unsupported combination of column-type and aggregation.");
+    }
+
+    if constexpr (cudf::is_fixed_width<target_type>()) {
+      return cudf::make_empty_column(data_type{type_to_id<target_type>()});
+    }
+
+    if constexpr (op == aggregation::COLLECT_LIST) {
+      return cudf::make_lists_column(
+        0, make_empty_column(data_type{type_to_id<offset_type>()}), empty_like(input), 0, {});
+    }
+
+    return empty_like(input);
+  }
+};
+
+std::unique_ptr<column> empty_output_for_rolling_aggregation(column_view const& input,
+                                                             rolling_aggregation const& agg)
+{
+  // TODO: Ideally, for UDF aggregations, the returned column would match
+  //       the agg's return type. It currently returns empty_like(input), because:
+  //         1. This is the behaviour currently expected in Pandas.
+  //         2. The UDF's return type is expressed as `datatype`,
```

Comment:
I think it would be more accurate to state that we're returning `empty_like` because we need to do some further investigation on whether cudf Python can/will support UDFs on nested columns, but that if we can support that use case we will need to pass the return type down from numba-jitted function.

Reply:
Hey, Vyas. Thanks for the review. Which of the two points would you recommend modifying?

1. I'm basing `1.` above on  the CUDF Python test that currently expects `empty_like(input)`, for UDFs called on empty columns, and we're aiming to preserve status quo. (I should rephrase "Pandas" to "CUDF Python").
2. `2.` is based on the following [signature](https://github.com/rapidsai/cudf/blob/branch-21.06/cpp/include/cudf/aggregation.hpp#L290) for `make_udf_aggregation()`:
```c++
template <typename Base = aggregation>
std::unique_ptr<Base> make_udf_aggregation(udf_type type,
                                           std::string const& user_defined_aggregator,
                                           data_type output_type);
```
We have the option of extracting the `output_type` and returning `make_empty_column(output_type)`. But that would break `#1`, and would still not handle UDFs that return nested types (e.g. lists).

Reply:
Reworded. Apologies for the lost progress in CI. :/